### DISCUSSION TITLE
Add hash digest for verification

### DISCUSSION THREAD

**Discussion by zaqas** [shadow/shadow#3212] [Ideas]
It would be appreciated if users could verify integrity of Shadow simulator. Please include them both on the website and Github release page.

**Comment by stevenengler** [comment]
The hash digest of the source code archive files? GitHub dynamically creates the zip/tar.gz file on the releases page, and not the Shadow project. I don't think adding a digest would be useful as (1) we didn't create it, (2) GitHub may change how it generates the archive and then the digest wouldn't match, and (3) GitHub controls everything from the repository to the website to the code archive, so the Shadow project already puts complete trust in GitHub.

Is there any particular reason for wanting the digest of the source code archive? If you want to verify the code locally, the tagged releases in the git repository are signed, so you can verify them with @robgjansen's [gpg key](https://www.torproject.org/static/keys/robgjansen.txt). (@robgjansen is this the same key that you sign the commits with?)

**Reply by robgjansen** [reply]
>  (@robgjansen is this the same key that you sign the commits with?)

Yes I sign my commits and the release tags with the same GPG key.
